stats/stats.go

```diff
@@ -178,36 +178,42 @@ func readStats() {
 		case AUTH_REQ_IN:
 			logger.StatsLog.Infoln("Received Event: AUTH_REQ_IN: ", m)
 			t := popTrans(m.Id) // remove MSG in trans but use the time msg was received
-			m.T = t.T
-			ue := getUe(m.Supi)
-			ue.CReg.AuthReqInTime = m.T
-			x := m.T.Sub(ue.CReg.RegReqOutTime)
-			ue.CReg.RegReqAuthReq = x.Microseconds()
-			logger.StatsLog.Infoln("Time[us] between Reg Req & Auth Req ", ue.CReg.RegReqAuthReq)
+			if t != nil {
+				m.T = t.T
+				ue := getUe(m.Supi)
+				ue.CReg.AuthReqInTime = m.T
+				x := m.T.Sub(ue.CReg.RegReqOutTime)
+				ue.CReg.RegReqAuthReq = x.Microseconds()
+				logger.StatsLog.Infoln("Time[us] between Reg Req & Auth Req ", ue.CReg.RegReqAuthReq)
+			}
```

Comment:
What about just adding the following right after `t := popTrans(m.Id)` (similar to the approach we suggested in the AMF PR)
```go
			if t == nil {
				continue
			}
```
```suggestion
			if t == nil {
				continue
			}
			m.T = t.T
			ue := getUe(m.Supi)
			ue.CReg.AuthReqInTime = m.T
			x := m.T.Sub(ue.CReg.RegReqOutTime)
			ue.CReg.RegReqAuthReq = x.Microseconds()
			logger.StatsLog.Infoln("Time[us] between Reg Req & Auth Req ", ue.CReg.RegReqAuthReq)
```

Reply:
Any comment/update about this suggestion?

Reply:
@thakurajayL, any comment/update about this suggestion?